### ISSUE TITLE
Break 1000+ LOC files into separate modules

### ISSUE THREAD

**Issue by Mrazator** [microsoft/vscode-eslint#1230]
Hey!

I am seeing a trend, in open source JS projects like this one, to keep and maintain files with more than 1000 lines of code.

As a potential contributor, it seems quite hard to follow this, as it is needed to go through the whole file and understand all of the thousands lines of code. 

How does this work you? Why are you following this trend? Did you think about giving it a structure and splitting it into some standalone modules? 

Thanks for reply!

**Comment by dbaeumer** [comment]
There is no deeper reason for this. It simply evolved that way.
